src/solc_0.8/defi/rules/RequirementsRules.sol

```diff
@@ -0,0 +1,280 @@
+        uint256 maxAmountId;
+        uint256 index;
+    }
+
+    mapping(IERC721 => RequirementRuleERC721) internal _listERC721;
+    mapping(IERC1155 => RequirementRuleERC1155) internal _listERC1155;
+    IERC721[] internal _listERC721Index;
+    IERC1155[] internal _listERC1155Index;
+
+    event ERC1155RequirementAdded(
+        address indexed contractERC1155,
+        uint256[] ids,
+        uint256 minAmountId,
+        uint256 maxAmountId
+    );
+    event ERC721RequirementAdded(
+        address indexed contractERC721,
+        uint256[] ids,
+        bool balanceOf,
+        uint256 minAmountBalanceOf,
+        uint256 maxAmountBalanceOf,
+        uint256 minAmountId,
+        uint256 maxAmountId
+    );
+    event MaxStakeOverallSet(uint256 newMaxStake, uint256 oldMaxStake);
+    event ERC1155RequirementDeleted(address indexed contractERC1155);
+    event ERC721RequirementDeleted(address indexed contractERC721);
+
+    modifier _isContract(address account) {
+        require(account.isContract(), "RequirementsRules: invalid address");
+
+        _;
+    }
+
+    modifier checkRequirements(
+        address account,
+        uint256 amount,
+        uint256 balanceOf
+    ) {
+        uint256 maxAllowed = _returnMaxStakeAllowed(account);
+
+        if (maxAllowed != 0) {
+            require(amount + balanceOf <= maxAllowed, "RequirementsRules: maxAllowed");
+        }
+
+        _;
+    }
+
+    // if user has not erc721 or erc1155
+    function setMaxStakeOverall(uint256 newMaxStake) external onlyOwner {
+        uint256 oldMaxStake = maxStakeOverall;
+        maxStakeOverall = newMaxStake;
+
+        emit MaxStakeOverallSet(newMaxStake, oldMaxStake);
+    }
+
+    function getMaxStakeAllowed(address account) external view returns (uint256) {
+        require(account != address(0), "RequirementsRules: invalid address");
+
+        return _returnMaxStakeAllowed(account);
+    }
+
+    function setERC721ListRequirement(
+        address contractERC721,
+        uint256[] memory ids,
+        bool balanceOf,
+        uint256 minAmountBalanceOf,
+        uint256 maxAmountBalanceOf,
+        uint256 minAmountId,
+        uint256 maxAmountId
+    ) external onlyOwner _isContract(contractERC721) {
+        IERC721 newContract = IERC721(contractERC721);
+
+        if (ids.length != 0) {
+            _listERC721[newContract].ids = ids;
+        }
+        _listERC721[newContract].minAmountBalanceOf = minAmountBalanceOf;
+        _listERC721[newContract].maxAmountBalanceOf = maxAmountBalanceOf;
+        _listERC721[newContract].minAmountId = minAmountId;
+        _listERC721[newContract].maxAmountId = maxAmountId;
+        _listERC721[newContract].balanceOf = balanceOf;
+
+        // if it's a new member create a new registry, instead, only update
+        if (_islistERC721Member(newContract) == false) {
+            _listERC721Index.push(newContract);
+            _listERC721[newContract].index = _listERC721Index.length - 1;
+        }
```

Comment:
make it a function (can be used by setERC721ListRequirement and setERC1155ListRequirement)

Reply:
won't change this atm